### PR TITLE
Assign owners to periodic and slow jobs

### PR DESCRIPTION
**Pull request by huydhn**
As an outcome of https://fburl.com/gdoc/voce5o06, I want to assign owner(s) to any periodic or slows job that are still needed but couldn't run more frequently (too $$$, capacity constraint, don't fail that often).  They include:

* multigpu
* debug build
* ROCm (distributed, slow)

@malfet @soulitzer I put down your names as the owners of debug build and slowgradcheck respectively.  Please let me know if you are ok with that, or if you have a better option in mind.

Any jobs there without an owner are owned by us (PT Dev Infra)

### Testing

The owners are show up in the job name https://hud.pytorch.org/pr/139519

### CODE REVIEW

File: .github/workflows/periodic.yml [hunk 1]
```diff
@@ -145,11 +145,11 @@ jobs:
       build-with-debug: true
       test-matrix: |
         { include: [
-          { config: "default", shard: 1, num_shards: 5, runner: "${{ needs.get-label-type.outputs.label-type }}linux.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 2, num_shards: 5, runner: "${{ needs.get-label-type.outputs.label-type }}linux.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 3, num_shards: 5, runner: "${{ needs.get-label-type.outputs.label-type }}linux.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 4, num_shards: 5, runner: "${{ needs.get-label-type.outputs.label-type }}linux.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 5, num_shards: 5, runner: "${{ needs.get-label-type.outputs.label-type }}linux.4xlarge.nvidia.gpu" },
+          { config: "default", shard: 1, num_shards: 5, runner: "${{ needs.get-label-type.outputs.label-type }}linux.4xlarge.nvidia.gpu", owners: ["malfet"] },
```
Comment (malfet) [inline .github/workflows/periodic.yml:148]: Do you mind adding some core oncall? Or may be add Alban's name in addition to mine?

File: .github/workflows/slow.yml [hunk 2]
```diff
@@ -58,14 +58,14 @@ jobs:
...
         { include: [
-          { config: "default", shard: 1, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 2, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 3, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 4, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 5, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 6, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 7, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu" },
-          { config: "default", shard: 8, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu" },
+          { config: "default", shard: 1, num_shards: 8, runner: "linux.g5.4xlarge.nvidia.gpu", owners: ["soulitzer"] },
```
Comment (malfet) [inline .github/workflows/slow.yml:61]: This does not sound right, owners should be a label on github rather than an individual user...

**Review by malfet — APPROVED** [review]

File: .github/workflows/slow.yml (hunk 2 above)
Comment (huydhn) [reply .github/workflows/slow.yml:61]: Switching this to `module: slowgradcheck`

Comment (huydhn) [reply .github/workflows/slow.yml:61]: Also update https://github.com/pytorch/pytorch/issues/24422#issue-481277499 to cc @soulitzer when this label is used.  This should give us the flexibility of updating the list later on

File: .github/workflows/periodic.yml (hunk 1 above)
Comment (huydhn) [reply .github/workflows/periodic.yml:148]: I have add a new `oncall: debug-build` into https://github.com/pytorch/pytorch/issues/24422 with you and Alban in the list, let me use that here instead

**Pull request closed without merging**